aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -98,124 +108,275 @@ module AptosFramework::Stake {
+        move_to(account, ValidatorSet {
+            consensus_scheme: 0,
+            minimum_stake,
+            maximum_stake,
+            active_validators: Vector::empty(),
+            pending_active: Vector::empty(),
+            pending_inactive: Vector::empty(),
+        });
     }
 
-    /// Withdraw from inactive delegation.
-    public(script) fun withdraw(account: &signer, from: address): Coin<TestCoin> acquires StakePool {
-        let addr = Signer::address_of(account);
-        let stake_pool = borrow_global_mut<StakePool>(from);
-        let delegation = withdraw_internal(&mut stake_pool.inactive, addr);
-        let Delegation {coins, rewards, locked_until_secs: _} = delegation;
-        Coin::merge<TestCoin>(&mut coins, rewards);
-        coins
+    /// This is only called during Genesis, which is where MintCapability<TestCoin> can be created.
+    /// Beyond genesis, no one can create TestCoin mint/burn capabilities.
+    public(friend) fun store_test_coin_mint_cap(core_resources: &signer, mint_cap: MintCapability<TestCoin>) {
+        move_to(core_resources, TestCoinCapabilities { mint_cap })
     }
 
-    /// Initialize the ValidatorInfo for account.
+    /// Initialize the validator account and give ownership to the signing account.
     public(script) fun register_validator_candidate(
         account: &signer,
         consensus_pubkey: vector<u8>,
         network_address: vector<u8>,
-        fullnode_address: vector<u8>
+        fullnode_address: vector<u8>,
     ) {
+        let account_address = Signer::address_of(account);
+        assert!(!exists<StakePool>(account_address), EALREADY_REGISTERED);
+
         move_to(account, StakePool {
-            voting_power: 0,
-            next_epoch_voting_power: 0,
-            active: IterableTable::new(),
-            pending_active: IterableTable::new(),
-            pending_inactive: IterableTable::new(),
-            inactive: IterableTable::new(),
+            active: Coin::zero<TestCoin>(),
+            pending_active: Coin::zero<TestCoin>(),
+            pending_inactive: Coin::zero<TestCoin>(),
+            inactive: Coin::zero<TestCoin>(),
+            locked_until_secs: 0,
+            operator_address: account_address,
         });
         move_to(account, ValidatorConfig {
             consensus_pubkey,
             network_address,
             fullnode_address,
         });
+        move_to(account, OwnerCapability {
+           pool_address: account_address,
+        });
+    }
+
+    /// Extract and return owner capability from the signing account.
+    public(script) fun extract_owner_cap(account: &signer): OwnerCapability acquires OwnerCapability {
+        move_from<OwnerCapability>(Signer::address_of(account))
+    }
+
+    /// Deposit `owner_cap` into `account`. This requires `account` to not already have owernship of another
+    /// staking pool.
+    public(script) fun deposit_owner_cap(account: &signer, owner_cap: OwnerCapability) {
```

Comment:
(script) won't work here as we cannot pass in structs.

Reply:
Updated